### PR TITLE
Only double precise scroll deltas for gesture devices (fix high-res mouse runaway scroll)

### PR DESCRIPTION
**Pull request by samuelpatro**
## Summary

The terminal scroll handler (`GhosttyNSView.scrollWheel`) doubles every precise scroll delta:

```swift
let precision = event.hasPreciseScrollingDeltas
if precision { x *= 2; y *= 2 }
```

macOS reports `hasPreciseScrollingDeltas == true` for **both** trackpads/Magic Mouse **and** high-resolution mice (e.g. Logitech free-spin / MX wheels). So a high-res mouse gets the 2x boost stacked on top of the acceleration macOS already applies, which makes scrolling feel runaway.

This was confirmed empirically by instrumenting `scrollWheel` with a real Logitech wheel: every event reported `precise=1` with `phase=0 momentum=0`, and a fast spin produced `dy=103` that cmux doubled to `206` per event at ~15 events/sec. A trackpad, by contrast, always drives a continuous gesture `phase` / `momentumPhase`; a plain wheel leaves both empty.

## What changed

- Extract the boost decision into a pure, testable helper `GhosttyNSView.shouldDoublePreciseScrollDelta(hasPreciseScrollingDeltas:phase:momentumPhase:)`, living in its own file `Sources/GhosttyTerminalScrollBoost.swift` so the already large `GhosttyTerminalView.swift` stays within the Swift file-length budget guard.
- Add a thin `shouldDoublePreciseScrollDelta(for:)` overload (also in the new file) so the `scrollWheel` call site stays a single readable line, while the decomposed overload remains for unit tests (a real `NSEvent` can't be synthesized in tests).
- Gate the 2x boost on `!phase.isEmpty || !momentumPhase.isEmpty`, so:
  - **Trackpad / Magic Mouse** (gesture phase present): unchanged, still boosted.
  - **High-res mouse** (precise deltas, no phase): no longer double-amplified, scrolls at the OS-accelerated rate.
  - **Notched mouse** (not precise): unchanged.
- The `precision` modifier bit sent to Ghostty is left intact (the deltas are still precise).

After the change the same fast spin sends ~103px/event instead of ~206px.

## Relationship to #5671 (`terminal.scrollSpeed`)

These two are complementary and stack into a good setup:

- **This PR** fixes the wrong *default* by correcting device classification, so high-res mice behave sanely out of the box with no configuration.
- **#5671** adds a user-tunable `terminal.scrollSpeed` multiplier (slider in Settings -> Terminal). With this PR merged, #5671 becomes pure fine-tuning on a correct baseline (e.g. nudge trackpad or mouse speed to taste) rather than the only escape hatch from the doubling.

Suggested sequencing: land this device-classification fix first, then #5671 on top so the multiplier applies to already-correct per-device deltas. The two changes touch the same handler but are otherwise independent.

## Tests

Two-commit red/green per repo policy:
- Commit 1 adds `GhosttyScrollPreciseBoostTests` plus the helper returning the old always-boost behavior; `testHighResMouseWithoutPhaseIsNotBoosted` fails (CI red).
- Commit 2 flips the helper to the phase-gated logic; all four cases pass (CI green).

Verified locally via `xcodebuild test -scheme cmux-unit -only-testing:cmuxTests/GhosttyScrollPreciseBoostTests` (4 passed). Test wired into `project.pbxproj` (lint-pbxproj-test-wiring passes).

## Localization

No user-facing strings added or changed; localization audit is a no-op for this PR.

https://claude.ai/code/session_01H4sYousP8J5cBAriRg9WEJ

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/6449"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1784482851&installation_id=133855650&pr_number=6449&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F6449&signature=25bde5b0efd5311764714423a9c30f4f202ea77ef84c8be376cc435cc162780f"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

---
## Summary by cubic
Stop doubling precise scroll deltas for high‑resolution mice. The 2x boost now applies only to gesture devices (trackpads/Magic Mouse) detected via scroll phase/momentum, fixing runaway scroll on Logitech free‑spin/MX wheels.

- **Bug Fixes**
  - Gate the 2x precise‑delta boost on non‑empty `phase` or `momentumPhase`.
  - High‑res mice (precise deltas, no phase) now scroll at the OS‑accelerated rate; trackpads remain boosted.
  - Added regression tests in `GhosttyScrollPreciseBoostTests` to cover gesture vs wheel cases.

- **Refactors**
  - Extracted the boost decision into `GhosttyTerminalScrollBoost` with an `NSEvent` initializer and `GhosttyNSView.shouldDoublePreciseScrollDelta(for:)`; simplified the call site.

<sup>Written for commit ffa202d9578b8e3b47e8307bfc06b34fd6639302. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/6449?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Improved scroll wheel/trackpad handling by applying the legacy precise-scroll boost only for gesture-like events, avoiding over-acceleration for mouse-style wheel input.
* **Bug Fixes**
  * Refines when precise scrolling deltas receive boosting, honoring event phase and momentum information to better match expected device behavior.
* **Tests**
  * Added regression coverage for multiple precise-scroll scenarios to ensure consistent boosting behavior across trackpad and mouse inputs.

### CODE REVIEW

**Comment by samuelpatro** [comment]
Pushed structural cleanups on top of the original red/green commits (no behavior change):

- **Fixed the failing `workflow-guard-tests` check.** The helper had pushed `GhosttyTerminalView.swift` (~11.8k lines) past the Swift file-length budget. Moved `shouldDoublePreciseScrollDelta(...)` into its own file `Sources/GhosttyTerminalScrollBoost.swift` as a `GhosttyNSView` extension, bringing the view file back to exactly budget with zero added debt.
- **Kept the call site readable.** Added a thin `shouldDoublePreciseScrollDelta(for:)` overload so `scrollWheel` reads as one line, while the decomposed `(hasPreciseScrollingDeltas:phase:momentumPhase:)` overload stays for the unit tests.
- **Wired the new file into `project.pbxproj`** and ran `normalize-pbxproj.py`.

Verified locally: budget guard exits 0, `lint-pbxproj-test-wiring` and `check-pbxproj` pass, and both `GhosttyTerminalScrollBoost.swift` and `GhosttyTerminalView.swift` compile clean (objects regenerate, no Swift errors). The runtime test still runs on CI.
